plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -266,10 +282,13 @@ func (c *MaxPDVolumeCountChecker) predicate(pod *v1.Pod, meta algorithm.Predicat
 
 	// randomPrefix is a prefix of auxiliary volume IDs when we don't know the
 	// real volume ID, e.g. because the corresponding PV or PVC was deleted. It
-	// is random to avoid conflicts with real volume IDs and it needs to be
-	// stable in whole predicate() call so a deleted PVC used by two pods is
-	// counted as one volume and not as two.
-	randomPrefix := rand.String(32)
+	// is generated by appending an integer (sequential) to a constant string
+	// (randomly generated during the checker's initialization) in order to
+	// avoid conflicts with real volume IDs and it needs to be stable in whole
+	// predicate() call so a deleted PVC used by two pods is counted as one volume
+	// and not as two. Also, appending a counter to a const string removes
+	// the need to create a random string each time (which is expensive).
+	randomPrefix := c.getNextVolumeIDPrefix()
```

Comment:
I'm not sure if a random prefix is needed for each predicate call.  It may be good enough to just generate a random prefix once during initialization.  All the structures used to count volumes in this predicate are local and shouldn't conflict if the predicate is running in parallel.

Reply:
That's precisely what I'm doing in this PR - a random prefix initialized once and then just appending it to a counter. Do you mean sth else?

Reply:
I mean, I don't think you even need a counter.  Just a random prefix initialized in the beginning.

Reply:
In that case do we even need the initialized string to be random? For e.g why not just set it to some hard-coded value like 'foobar'?

Reply:
don't you risk conflicting with actual object names?

Reply:
Yes, we just want a prefix that won't conflict with actual object names.  So we could generate some random prefix, or have some hardcoded value that we are sure nobody would ever use.  I don't think it needs to be generated every time we run the predicate though.

Reply:
Changed it - PTAL.